bastionzero/service/targets_disambiguated/all_targets.go

```diff
@@ -0,0 +1,132 @@
+package targets_disambiguated
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/service/agents"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/service/connections/connectionstate"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/service/targets/targetstatus"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/types"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/types/targettype"
+	"github.com/bastionzero/bastionzero-sdk-go/internal/client"
+)
+
+const (
+	targetsBasePath = "api/v2/targets"
+)
+
+// AllTargetsService handles communication with the disambiguated targets endpoints
+// of the BastionZero API.
+//
+// BastionZero API docs:
+//
+// - https://cloud.bastionzero.com/api
+
+type AllTargetsService client.Service
+
+type AccessDetails struct {
+	JIT                  bool            `json:"jit"`
+	AccessExpirationTime types.Timestamp `json:"accessExpirationTime"`
+}
+
+// Port describes a port number
+type Port struct {
+	Value *int `json:"value,omitempty"`
+}
+
+// TargetInterface lets you work with common attributes from any kind of
+// BastionZero target
+type TargetInterface interface {
+	// GetID returns the target's unique ID.
+	GetID() string
+	// GetName returns the target's name.
+	GetName() string
+	// GetStatus returns the target's status.
+	GetStatus() targetstatus.TargetStatus
+	// GetEnvironmentID returns the target's environment's ID.
+	GetEnvironmentID() string
+	// GetLastAgentUpdate returns the target's last agent update if set.
+	// Otherwise, returns nil.
+	GetLastAgentUpdate() *types.Timestamp
+	// GetAgentVersion returns the target's agent version.
+	GetAgentVersion() string
+	// GetRegion returns the target's region.
+	GetRegion() string
+	// GetAgentPublicKey returns the target's agent public key.
+	GetAgentPublicKey() string
+	// GetTargetType returns the target's type.
+	GetTargetType() targettype.TargetType
+}
```

Comment:
Doesn't look like this is being used by the new `_target.go` files in `targets_disambiguated`. Feel free to remove. In a future PR, we can bring this interface back and add common getters for the new disambiguated types.